clients/cobol-lsp-vscode-extension/src/services/JavaCheck.ts

```diff
@@ -14,14 +14,21 @@
 import * as cp from "child_process";
 import { SettingsService } from "./Settings";
 
-const versionPattern = new RegExp(
-  '(java|openjdk) (version)? ?"?((9|[0-9][0-9])|(1|9|[0-9][0-9]).(1|8|[0-9][0-9]).*).*',
-);
+export const SUPPORTED_JAVA_VERSION = 11;
+
+const versionCapturer =
+  /\b(?:java|openjdk)\b(?:\s+version)?\s+"?(?:1\.)?(\d{1,2})(?:\.\d+)*[-_A-Za-z0-9]*"?/i;
```

Comment:
Wouldn't the following be enough?
```suggestion
  /\b(?:java|openjdk)\b(?:\s+version)?\s+"?(?:1\.)?(\d+)/i;
```